content(cms): create Post "1-einführung-markup-sprachen-von-html-zu-tei/index" (#34)
acdh-oeaw/howto: content/posts/1-einführung-markup-sprachen-von-html-zu-tei/index.mdx

> ---

Reviewer: Neue Bilder werden als Issue hinterlegt, damit das nicht vergessen wird.
Link könnte man einfügen, ich würde, sobald dann die Teile zusammengebunden sind, vielleicht noch einen Absatz in der Curriculums-Oberfläche (noch nicht vorhanden) einfügen, wo wir den Aufbau beschreiben (und dort dann den Link einfügen)

> ---

Reviewer 2: Zu Bildern: Rechtschreibung müssten wir auch anpassen - da im Text immer "Paragrafen" steht, sollte das auch auf der Abbildung "XML Struktur eines Buches" so sein (Zeile 292).

> ---

Reviewer 3: ![Folie_11](https://user-images.githubusercontent.com/52656595/134899803-4818611e-ed84-4391-9190-f0ce2573e9fb.PNG)
Anbei die ausgebesserte Folie

> ---

Author: Danke Kiki! Ich habe das Bild (wie auch die anderen mit Rechtschreibkorrektur) getauscht.

> title: 1. Einführung Markup-Sprachen - von HTML zu TEI

Reviewer 3: ```suggestion
title: 1. Markup-Sprachen: von HTML zu TEI
```

> title: 1. Einführung Markup-Sprachen - von HTML zu TEI

Reviewer 3: => Ist zwar resolved, bin aber nicht einverstanden ...

Reviewer 2: Sorry, da ist mir ein Fehler passiert - hab anscheinend zu früh auf resolved geklickt. ich schau, dass ich das noch ausbessere...

lang: de
date: 2021-07-23T10:36:13.905Z
version: 1.0.0
authors:
  - czeitschner-ulrike
  - schopper-daniel
editors:
  - königshofer-elisabeth
  - wiencek-florian
tags:
  - tei
  - xml

> abstract: von XML zu TEI

Reviewer 3: ```suggestion
abstract: Dieser Abschnitt gibt einen Überblick über den Digitalisierungsprozess und die wichtigsten die Markup-Sprachen vor.
```

licence: ccby-4.0
toc: true
uuid: ycJztJbuEaDJyNFBehyEe
---

> ## Lernziele

Reviewer: würde das dann generell als Standardausdruck ausdehnen auf alle Module

> ## Lernziele

Reviewer 3: Werde ich überall ändern, hoffe ich übersehe nichts.

- Überblick über den Digitalisierungsprozess vom physischen Originaldokument zu digitaler Annotation
- Kennenlernen von XML-Elementen, -Syntax, -Dokumentmodell, und -Hierarchie
- Definition der Begriffe Wohlgeformtheit und Validität

> - Definition von Schema und Namensraum

Reviewer 3: ```suggestion
- Schema und Namensraum
```

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer: vielleicht hier ergänzen:
TEI - ein Dokumenten- und Auszeichnungsformat für die Textwissenschaften

Bei Dokumentenformat denke ich an Dinge wie .docx, .pdf etc.

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer 3: Was hältst Du von:
TEI – Dokumentenformat und Markup-Sprache für die Textwissenschaften

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer: Gefällt mir auf jeden Fall besser, weil konkreter

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer: @SusanneZhanial ? @fwiencek ?

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer 2: Ich als Neuling: wie Elisabeth denke ich beim Wort Dokumentenformat zuerst mal an .docx, .pdf o.ä. - trifft das auf TEI auch zu? Wie wäre es, wenn man die Begriffe umdreht: 
TEI - Markup-Sprache und Dokumentenformat für die Textwissenschaften

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer 3: Ja, umgedreht geht auch, ist sogar besser! Der Wikipedia-Artikel sagt: TEI ist eine Organisation und ein gleichnamiges Dokumentenformat ... hat zwar keine eigene Endung, ist aber eben nicht nur Markup-Sprache, sondern zugleich auch Dokumentenformat: https://www.wikiwand.com/de/Text_Encoding_Initiative
**Was anderes:** Bitte bei **Bindestrichen** "–" und nicht"-" verwenden. (Habe das hier https://oeawacat.sharepoint.com/:w:/r/sites/ACDH-CH_p_DiTAH_DigitaleTransformationsterreichischeGeistesw/_layouts/15/doc2.aspx?action=edit&sourcedoc=%7B4FC6DE52-75C6-4C95-82DF-33C58BAACDC3%7D dokumentiert)

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Author: I second Susanne -  TEI - Markup-Sprache und Dokumentenformat für die Textwissenschaften

Infos zu XML als Dokumentenformat gibt es auch hier: https://www.data2type.de/xml-xslt-xslfo/xml/xml-in-a-nutshell/xml-als-dokumentformat

> - Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

Reviewer 3: => Letztstand im Vorschlag eingefügt.

## Wie kommt ein Text in ein digitales Format?

Sofern es sich nicht um ein born digital-Dokument handelt - also ein Dokument das mit einem Computer erstellt wurde und deswegen bereits in digital verarbeitbarer Form vorliegt -, beginnt die

> Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

Reviewer 3: Es ist nur deshalb blöd, weil wenn wir nicht auf unser Digitalisierungszentrum verweisen, wird offensichtlich, dass es hier bei uns den Mangel gibt, dass wir online dazu nix zum herzeigen haben ... deshalb bin ich da lieber vorsichtig. Wir sollten aber anregen, dass dieser Mangel rasch behoben werden sollte aus den eben genannten Gründen.

> Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

Author: Wichtig ist es mir hierbei nicht nur das Scannen oder den Scanner an sich zu beleuchten, sondern das Thema der Digitalisierung. Dies finde ich im Artikel der Uni Trier im Hinblick auf Textquellen sehr gut beschrieben. Zudem verweist der Text auf die DFG-Praxisregeln »Digitalisierung«, online: http://www.dfg.de/formulare/12_151/12_151_de.pdf, die sich gut als weiterführende Literatur eigenen. 

Einen reinen Foliensatz ohne den dazugehörigen Vortrag halte ich als weiterführende Literatur etwas zu kurz gegriffen. 

Im Hinblick auf eigene Quellen: es ist finde ich gerade das Wesen des Internets, unterschiedliche Ressourcen miteinander zu verlinken. Hier finde ich es auch nicht schlimm, wenn wir zunächst keine "eigene" Ressource zu dem Thema haben wenn es eine adäquate Ressource im Netz bereits gibt. Ziel des HowTo ist es ja gerade Ressourcen zu entwickeln und vor allem auch zu kontextualisieren und kuratieren. Insofern würde ich als Leser von außen hier nichts vermissen.

Für die Internen Leser*innen wäre natürlich ein Verweis auf unser Digitalisierungszentrum interessant. Bis jetzt habe ich im Netz zwei Dinge gefunden: https://www.oeaw.ac.at/acdh/what-we-offer/during-the-project, https://forschungsinfrastruktur.bmbwf.gv.at/de/fi/acdh-digitalisierungszentrum_3894

> Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

Reviewer 3: Ich stimme Dir voll und ganz zu, würde aber selbst die Verantwortung für die Auswahl übernehmen wollen, sondern zumindest Rücksprache mit Charly und Matej halten, ob sie damit grundsätzlich einverstanden sind. Ist das nachvollziehbar?

> Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

Reviewer 3: Bitte den Text mal so wie er ist, einfügen - die Links können wir beim letzten Korrekturdurchgang gesammelt klären.

> Deswegen erfolgt daran anschließend in einem zweiten Schritt die **Texterkennung mittels ["optical character recognition" (OCR)](https://de.wikipedia.org/wiki/Texterkennung)**. Das Ergebnis ist ein elektronischer Text - z.B. in Form eines Word-Dokuments - oder einer Text-Datei, den man zwar mit verschiedenen Programmen strukturieren und formatieren kann, wobei die Möglichkeiten sowohl im Hinblick auf das Layout als auch die inhaltliche Beschreibung meistens sehr begrenzt sind. Außerdem ist die Konvertierung in andere Formate oft nicht verlustfrei möglich, wenn die Formatierungs- oder Strukturinformationen in programmspezifischen Formaten abgespeichert sind. ([\> mehr Informationen zu Volltextdigitalisierung](https://dhmuseum.uni-trier.de/node/49))

Reviewer 2: Ich konnte keine österreichische Quelle finden; es ist meistens Uni Trier oder das Buch " Standards und Methoden der Volltextdigitalisierung. Beiträge des Internationalen Kolloquiums an der Universität Trier", die man findet.

Um elektronische Daten systematisch, am besten mehrfach und interdisziplinär für sprachwissenschaftliche, literaturwissenschaftliche oder kulturwissenschaftliche Forschung nutzen zu können, benötigen sie **dauerhafte,** strukturelle, semantische und linguistische **Annotationen**. An diesem Punkt kommen sogenannte Auszeichnungs- oder Markup-Sprachen zum Einsatz. 

## Markup-Sprachen

Auszeichnungssprachen dienen der Beschreibung von unterschiedlichsten Daten z.B. von Texten, Bildern, Filmen usw. Gleichzeitig dienen sie auch der Beschreibung des Verfahrens, das zu ihrer Weiterverarbeitung bzw. den Datenaustausch notwendig ist. Eine **Markup-Sprache** ist also im Wesentlichen eine Strukturierungssprache, die es erlaubt, maschinenlesbar die Gliederung und Formatierung von u.a. Texten auszuzeichnen. 

> Eine der bekanntesten "Markup Languages" (ML) ist die "[Hypertext Markup Language" (HTML)](https://www.w3.org/html/), die eine der technologischen Grundlagen des World Wide Webs bildet. Sie gliedert die multimodalen Inhalte einer Webseite so, dass ein Webbrowser aber auch Suchmaschinen und Algorithmen diese interpretieren und darstellen können. Hierfür kommt ein beschränktes Vokabular an vordefinierten "Tags" zum Einsatz, das in Web-Standards und Richtlinien des[ World Wide Web Consortiums (W3C)](https://www.w3.org/) festgelegt wurde und von standardkonformen Webbrowsern unterstützt wird. Im modernen Web-Publikationen werden Inhaltsstruktur bzw. strukturelle Semantik (z.B. Überschriftenebenen, Absätze, Listen, Hyperlinks, Zitate, und andere Medienelemente oder interaktive Elemente) und deren Darstellung getrennt. HTML übernimmt dabei die logische Strukturierung der Inhalte, während andere Technologien wie [Cascading Stylesheets (CSS)](https://www.w3.org/Style/CSS/Overview.en.html) oder [JavaScript](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/) für die Darstellung und die Interaktion mit den Inhalten zum Einsatz kommen.

Author: Man kann hier meiner Meinung nach auch "Die bekannteste ML" schreiben... Widersprechen muss ich allerdings was die technischen Grundlagen des WWW anbelangt. Die technischen Grundlagen im Web bestehen nicht nur aus einer ML, sondern es braucht auch ein zugehöriges Protokoll (in dem Fall http), nebst der physischen Infrastruktur des Internet (WWW ist nur ein Service das auf dem Internet basiert). Insofern wäre es nicht ganz richtig das außen vor zu lassen. Entwickelt wurde die Technologie (HTML + HTTP) von Tim Berners-Lee bereits 1989 am CERN, Standardisierung dann später über W3C.

Im Bezug auf das alte WWW ist auch multimedial passend da hier andere "Modalitäten" noch keine so große Rolle spielen. Insofern kann man den Begriff in dem speziellen Fall interchangably verwenden.

Webbrowser stellen dar, Algorithmen (auch die der Suchmaschinen) nutzen die strukturierten Inhalte allerdings für die Suche und dem Verständnis, um was es geht. Die Strukturierung einer Seite spiel z.B. für SEO eine große Rolle (neben anderen Faktoren). Das Finden leitet sich also u.a. aus dem Interpretieren und Maschinenverständnis ab.

Bei linguistischen Fachtermini bin ich nicht der Experte, da vertraue ich euerer Einschätzung - wenn ihr hier noch etwas linguistisch Ergänzen wollt go ahead. Die zweite Hälfte kann grundsätzlich so bleiben, allerdings würde ich tatsächlich die Trennung von Struktur und Darstellung hervorheben.

JavaScript ermöglicht allerdings nicht nur die Einbettung von interaktiven Elementen sondern die Programmierung der Client-seitigen Interaktion. Einbettung geht hier als Begriff nicht weit genug. Vielleicht eher Integration statt Einbettung?

> Eine der bekanntesten "Markup Languages" (ML) ist die "[Hypertext Markup Language" (HTML)](https://www.w3.org/html/), die eine der technologischen Grundlagen des World Wide Webs bildet. Sie gliedert die multimodalen Inhalte einer Webseite so, dass ein Webbrowser aber auch Suchmaschinen und Algorithmen diese interpretieren und darstellen können. Hierfür kommt ein beschränktes Vokabular an vordefinierten "Tags" zum Einsatz, das in Web-Standards und Richtlinien des[ World Wide Web Consortiums (W3C)](https://www.w3.org/) festgelegt wurde und von standardkonformen Webbrowsern unterstützt wird. Im modernen Web-Publikationen werden Inhaltsstruktur bzw. strukturelle Semantik (z.B. Überschriftenebenen, Absätze, Listen, Hyperlinks, Zitate, und andere Medienelemente oder interaktive Elemente) und deren Darstellung getrennt. HTML übernimmt dabei die logische Strukturierung der Inhalte, während andere Technologien wie [Cascading Stylesheets (CSS)](https://www.w3.org/Style/CSS/Overview.en.html) oder [JavaScript](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/) für die Darstellung und die Interaktion mit den Inhalten zum Einsatz kommen.

Reviewer 3: => Letztstand im Vorschlag eingefügt.

      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider nicht ganz richtig!**

        TEI ist eine fachspezifische Markup-Sprache, die auf XML basiert und dafür ausgelegt ist, Dokumente für textwissenschaftliche Forschung zu annotieren.

        Andere Markup-Sprachen sind HTML und XML. Während HTML allerdings ein festes Vokabular anbietet, das für die Strukturierung von Inhalten für Webpublikationen ausgelegt ist, stellt XML als ein abstraktes Werzeug zur Modellierung und Strukturierung von Informationen lediglich Regeln bereit, wie Elemente zu definieren und anzuwenden sind. Man kann aber beliebig viele konkrete Auszeichnungssprachen mithilfe von XML definieren.
      </Quiz.Message>
    </Quiz.MultipleChoice>
  </Quiz.Card>
</Quiz>

## Grundlagen von XML

> Wie schon erwähnt, dient XML dazu, beliebige Informationen und Inhalte in strukturierter Form zu beschreiben um diese Daten weiterzuverarbeiten, zu verwalten, oder sie im Internet auszutauschen. Eine – wenn nicht die wesentlichste – Eigenschaft von XML ist dabei die Unabhängigkeit von einzelner Hard- und Software. Möglich ist das, weil XML komplexe Inhalte beschreibt, ohne festzulegen, in welcher Form und mit welchen Computerprogrammen die Daten weiterverarbeitet werden sollen. Das bedeutet, es handelt sich um eine sogenannte Deskriptive und keine Processing Markup Language. XML ist so konzipiert, dass es von vielen unterschiedlichen Programmen gelesen werden kann, was eine enorme Freiheit gegenüber herstellergebundenen Formaten wie Microsoft Word, das ohne die entsprechende Software lange Zeit nicht verarbeitet werden konnte. Diese Eigenschaft war zugleich die wichtigste Voraussetzung für einen problemlosen und weltweiten Datenaustausch.

Reviewer 2: Frage: kann man Microsoft-Dateien inzwischen nicht auch mit alternativen Produkten (e.g. Open Office) öffnen? Ich glaube, dass es dahingehend so formuliert war (@fwiencek?)

> Wie schon erwähnt, dient XML dazu, beliebige Informationen und Inhalte in strukturierter Form zu beschreiben um diese Daten weiterzuverarbeiten, zu verwalten, oder sie im Internet auszutauschen. Eine – wenn nicht die wesentlichste – Eigenschaft von XML ist dabei die Unabhängigkeit von einzelner Hard- und Software. Möglich ist das, weil XML komplexe Inhalte beschreibt, ohne festzulegen, in welcher Form und mit welchen Computerprogrammen die Daten weiterverarbeitet werden sollen. Das bedeutet, es handelt sich um eine sogenannte Deskriptive und keine Processing Markup Language. XML ist so konzipiert, dass es von vielen unterschiedlichen Programmen gelesen werden kann, was eine enorme Freiheit gegenüber herstellergebundenen Formaten wie Microsoft Word, das ohne die entsprechende Software lange Zeit nicht verarbeitet werden konnte. Diese Eigenschaft war zugleich die wichtigste Voraussetzung für einen problemlosen und weltweiten Datenaustausch.

Author: Ja genau, @SusanneZhanial. Word kann mittlerweile auch mit Programmen von Drittherstellern geöffnet und bearbeitet werden (OpenOffice, Google Docs, etc.). Das Wordformat war lange proprietär, aber seit dem Wechsel zu einem XML-basierten Dateiformat ist es deutlich zugänglicher. Deswegen die Formulierung. Aber bei der Verwendung von Drittsoftware gibt es natürlich keine Garantie, dass alle Funktionen verwendbar sind oder sich die Darstellung nicht verändert (auch wenn das mittlerweile deutlich besser geworden ist).

> Wie schon erwähnt, dient XML dazu, beliebige Informationen und Inhalte in strukturierter Form zu beschreiben um diese Daten weiterzuverarbeiten, zu verwalten, oder sie im Internet auszutauschen. Eine – wenn nicht die wesentlichste – Eigenschaft von XML ist dabei die Unabhängigkeit von einzelner Hard- und Software. Möglich ist das, weil XML komplexe Inhalte beschreibt, ohne festzulegen, in welcher Form und mit welchen Computerprogrammen die Daten weiterverarbeitet werden sollen. Das bedeutet, es handelt sich um eine sogenannte Deskriptive und keine Processing Markup Language. XML ist so konzipiert, dass es von vielen unterschiedlichen Programmen gelesen werden kann, was eine enorme Freiheit gegenüber herstellergebundenen Formaten wie Microsoft Word, das ohne die entsprechende Software lange Zeit nicht verarbeitet werden konnte. Diese Eigenschaft war zugleich die wichtigste Voraussetzung für einen problemlosen und weltweiten Datenaustausch.

Reviewer 3: Ja genau, man kann Word sicherlich mit anderen Programmen öffnen usw., damit man aber garantiert verlustfrei weiterverarbeiten kann, annotiert man im softwarunabhängigen XML/TEI.

Die Bezeichnung von XML als Markup Language ist leider nicht sehr präzise gewählt, wenn nicht sogar irreführend. Denn XML  definiert keine Elemente. Es stellt vielmehr ein Regelwerk für den Entwurf von konkreten Auszeichnungssprachen und ihren Anwendungen dar und wird deshalb auch als Metasprache bezeichnet, die unbegrenzt erweiterbar ist. Das bedeutet, dass sich mit XML beliebig viele unterschiedliche Markup-Sprachen definieren lassen, die jeweils ganz unterschiedliche Zwecke verfolgen können. XML selbst ist leicht verständlich, da es nicht aus abstrakten Codes, sondern aus Text besteht, was den Vorteil hat, dass sie von Mensch und Maschine gelesen werden kann. Dies ist neben der Plattformunabhängigkeit und der Erweiterbarkeit mit ein Grund für die weite Verbreitung und den Erfolg von XML.

> Verantwortlich für die Entwicklung von XML ist das 1994 gegründete [World Wide Web Consortium](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.

Reviewer 2: Wäre hier eine Verlinkung im letzten Satz zur 5. Ausgabe sinnvoll bzw. möglich?

> Verantwortlich für die Entwicklung von XML ist das 1994 gegründete [World Wide Web Consortium](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.

Reviewer 3: Das ist wäre der allgemeine Link https://tei-c.org/release/doc/tei-p5-doc/en/html/index.html, den wir auch an der Stelle, an der die Module erwähnt werden, verwenden. Spricht nichts dagegen den hier einzufügen, oder?

> Verantwortlich für die Entwicklung von XML ist das 1994 gegründete [World Wide Web Consortium](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.

Author: Den Link kann man auch an dieser Stelle einfügen...

> Verantwortlich für die Entwicklung von XML ist das 1994 gegründete [World Wide Web Consortium](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.

Reviewer 3: => Letztstand im Vorschlag eingefügt.

```

*Richtig:*

```xml
<a>Inhalt <b>Inhalt</b></a> <b>Inhalt</b>
```

<SideNote type="note" title="Merke">
Die Regel der "well-formedness" ist zwingend, d.h. **alle XML-Dokumente müssen wohlgeformt sein!**

- Geöffnete Element müssen wieder geschlossen werden, oder leere Elemente sein. 
- Bei Verschachtelung muss die richtige Reihenfolge beim Öffnen und Schließen von Elementen eingehalten werden.
</SideNote>

### Validität

Zusätzlich dazu kann ein XML-Dokument auch valide oder gültig sein. Was bedeutet das? 

Wie schon erwähnt, kann im Grunde jeder mit XML seine eigene Markup-Sprache für bestimmte Zwecke definieren. Gleichzeitig ist es für wissenschaftliche Kollaboration und Open Science wichtig, die erzeugten Dokumente weitergeben und mehrfach nachnutzen zu können. **Wenn man also eine Sprache "erfindet", muss man nicht nur das Vokabular, sprich die Elemente, Attribute und Werte festlegen, sondern sollte auch eine Grammatik schreiben**, in der definiert ist, was die Elemente bedeuten, in welcher Beziehung sie zueinander stehen, wo und wie sie verwendet werden dürfen. 

Obwohl diese Regel optional ist, empfiehlt es sich, sie einzuhalten – aus mehreren Gründen: 

> - Eine Grammatik ist zum einen nützlich, damit man selbst überprüfen kann, ob man mehrere Dokumente desselben Typs einheitlich annotiert hat. 

Reviewer 3: ```suggestion
- Ein Schema ist zum einen nützlich, damit man selbst überprüfen kann, ob man mehrere Dokumente desselben Typs einheitlich annotiert hat. 
```

  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Welche der folgenden Aussagen ist richtig?**
      </Quiz.Question>

      <Quiz.MultipleChoice.Option>
        Ein XML Dokument muss valide sein. Idealerweise ist es auch wohlgeformt.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        Ein XML Dokument muss wohlgeformt sein. Es wird empfohlen durch die Nutzung und Überprüfung einer Grammatik auch die Validität des Dokuments zu gewährleisten.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>
        XML besitzt keinerlei Regelwerk. Wohlgeformtheit und Validität sind lediglich Empfehlungen.
      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">
        **Richtig!**

        Die Wohlgeformtheit eines Dokuments ist zwingend notwendig. Es wird aber empfohlen ein Dokument ebenso valide zu gestalten.
      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider falsch.**

        Die Wohlgeformtheit eines Dokuments ist zwingend notwendig. Es wird aber empfohlen ein Dokument ebenso valide zu gestalten.
      </Quiz.Message>
    </Quiz.MultipleChoice>
  </Quiz.Card>
</Quiz>

> ### Elementhierarchie in XML

Reviewer: Stimme zu, den würde ich auch runter ziehen.
Ich habe die Reihenfolge noch nicht geändert, da ich nicht sicher bin, ob dann nicht die Kommentare oben verschwinden. Ist derzeit auch ein "Issue" (bitte auflösen, sobald alle anderen erledigt sind).

> ### Elementhierarchie in XML

Reviewer 3: Ja bitte diesen Abschnitt noch verschieben und **nach** ### Aufbau von XML-Dokumenten und **vor** dem Quiz einreihen.

Ein XML-Dokument ist eine hierarchische Strukturierung und Anordnung von Inhalten. Diese ist jedoch nicht beliebig, sondern wird im Dokument-Schema festgelegt. Und da Dokumente gültig sein sollen, kann man sich die Strukturierung nicht willkürlich aussuchen.

**Das Schema schreibt vor, welche Elemente es gibt, wo sie anzutreffen sein dürfen und wo nicht.** So kann beispielsweise im Schema definiert werden, dass ein Element nur ein bestimmtes Elternelement haben kann.

- Ein Vorname darf nur das Kind eines Personennamens sein.
- Ein Personenname kann nur *innerhalb* einer Überschrift, eines Paragraphen oder einer Verszeile vorkommen, aber niemals außerhalb.

Neben den Beziehungen regelt ein Schema auch, **wie oft ein Element oder Attributwert vorkommen darf**. 

- Eindeutige ID-Nummern dürfen sinnvollerweise nur einmal vergeben werden. 
- Ebenso sollte in einem Geburtsregister jede Person nur ein Geburtsdatum haben.

Ein Buch, könnte in XML etwa wie folgt abgebildet werden: 

> ![XML Hierarchie eines Buches](images/xmlhierarchiebuch.png "XML Hierarchie eines Buches. Bild: Ulrike Czeitschner, ACDH-CH")

Reviewer 3: ```suggestion
![XML Hierarchie eines Buches](images/xmlhierarchiebuch.png "XML-Struktur eines Buches. Bild: Ulrike Czeitschner, ACDH-CH")
```

Innerhalb des Wurzelelements befindet sich ein Element mit dem Namen Buch, das die Kindelemente Vorwort und mehrere Kapitel enthält, vielleicht gefolgt von einem Nachwort, eventuell einem Register und einem Impressum. Diese Kindelemente verhalten sich zueinander alle wie Geschwister und sind aber selbst auch Eltern der Titel-, Paragraphen- und Verszeilen-Elemente, die ebenfalls Geschwister sind und selbstverständlich auch wieder Kinder haben können. Somit ergibt sich eine hierarchische Anordnung und Struktur der Inhalte, die den Aufbau des Buches widerspiegelt.

> ### Aufbau von XML-Dokumenten

Reviewer 3: ```suggestion
### Der Aufbau von XML-Dokumenten
```

> ![Aufbau XML Dokument](images/xml-dokument.png "Aufbau eines XML Dokuments. Bild: Ulrike Czeitschner, ACDH-CH")

Reviewer 3: ```suggestion
![Aufbau XML Dokument](images/xml-dokument.png "Aufbau eines XML-Dokuments. Bild: Ulrike Czeitschner, ACDH-CH")
```

Daran anschließend kann optional ein **Verweis auf dieses externe Schema** folgen und ebenso optional ein weiterer Verweis, der **processing instruction** (Verarbeitungsverweis) genannt wird und den Pfad zu einem sogenannten Stylesheet angibt, in dem die Verarbeitungsanweisungen definiert sind. Diese Anweisungen sind nicht Teil der aus Zeichen bestehenden Daten des XML-Dokuments, sondern dienen dazu Informationen und Anweisungen an Applikationen weiterzugeben, die diese Daten verarbeiten.

Nach dem Prolog folgen die eigentlichen Elemente. Wesentlich ist hier, dass es **nur ein einziges Wurzel- oder Root-Element** geben darf, das **alle anderen Elemente (Kinder und Kindeskinder) umschließt**. 

Im Wurzelelement befindet sich ein Attribut mit der Bezeichnung **namespace**, das als Wert einen Unified Ressource Identifier (URI) enthält, der angibt, woher die verwendeten Elemente und Attribute stammen. 

Handelt es sich z.B. um ein **Dokument, das nach den Richtlinien der TEI** erstellt wurde, ist der **Name des Root-Elements TEI**. Der Attributwert der namespace declaration verweist darauf, dass ausschließlich von der TEI definierte Elemente und Attribute verwendet wurden. 

> ![TEI Namensraum](images/tei-namensraum.png "TEI Namensraum. Bild: Ulrike Czeitschner, ACDH-CH")

Reviewer 3: ```suggestion
![TEI Namensraum](images/tei-namensraum.png "TEI-Namensraum. Bild: Ulrike Czeitschner, ACDH-CH")
```

Möchte man Komponenten aus verschiedenen Namensräumen mischen, müssen zwei Namenraum-Deklarationen angegeben werden, die jeweils unterschiedliche Präfixe definieren, z.B. ein Attribut xmlns:tei und ein Attribut xmlns:xyz. 

> Im Dokument selbst müssen diese Präfixe, gefolgt von einem Doppelpunkt, den betreffenden Elementnamen vorangestellt werden um deren Schema-Zugehörigkeit anzuzeigen.

Reviewer 3: ```suggestion
Im Dokument selbst müssen diese Präfixe, gefolgt von einem Doppelpunkt, den betreffenden Elementnamen vorangestellt werden um deren Schema-Zugehörigkeit anzuzeigen:
```

![mehrere Namensraeume](images/mehrerenamensraeumexml.png "Mehrere Namensräume. Bild: Ulrike Czeitschner, ACDH-CH")

<SideNote type="note" title="Hinweis">

> Kommentare sind außerhalb von einführenden oder schließenden Tags überall erlaubt. Sie werden eingeleitet von einer Spitzklammer, einem Rufzeichen, sowie zwei Minuszeichen. Dann folgt der Kommentartext und danach zwei Minuszeichen und eine schließende Spitzklammer.

Reviewer 3: ```suggestion
Kommentare sind außerhalb von einführenden oder schließenden Tags überall erlaubt. Sie werden eingeleitet von einer Spitzklammer, einem Rufzeichen, sowie zwei Minuszeichen. Dann folgt der Kommentartext und danach zwei Minuszeichen und eine schließende Spitzklammer:
```

```xml

> <!-- Kommentar -->

Reviewer 3: ```suggestion
<!-- Kommentar -->
```

      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>
        **Namensraum-Deklaration**: URI die angibt, woher die Elemente und Attribute stammen.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        **Verarbeitungsverweis**: enthält den Pfad zum Stylesheet mit Verarbeitungshinweisen.
      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">
        **Richtig!**

        Die XML-Deklaration, der Schema-Verweis und der Verarbeitungsverweis können Teil des Prologs einer XML-Datei sein.

        Die Namensraum-Deklaration ist Teil des Wurzelelements, das auf den Prolog folgt.
      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider nicht ganz richtig.**

        Die XML-Deklaration, der Schema-Verweis und der Verarbeitungsverweis können Teil des Prologs einer XML-Datei sein.

        Die Namensraum-Deklaration ist Teil des Wurzelelements, das auf den Prolog folgt.
      </Quiz.Message>
    </Quiz.MultipleChoice>
  </Quiz.Card>
</Quiz>

## Von XML zu TEI

> 1987 entwickelte die neu gegründete Text Encoding Initiative (TEI) ein spezialisiertes Schema für die Kodierung von Texten und sprachlichen Informationen.  Basierend auf XML ist TEI mittlerweile der de-facto-Standard für die Kodierung von Texten und sprachlichen Informationen und ist innerhalb der Geisteswissenschaften federführend.  Die derzeit empfohlene Version P5 wurde 2007 veröffentlicht und seitdem regelmäßig verbessert und aktualisiert. Das Regelwerk enthält mehr als 500 Elemente und ebenso zahlreiche Attribute, die in sachbezogenen Modulen organisiert sind. Sie berücksichtigen Lyrik und Dramentexte ebenso wie Handschriften, kritische Apparate oder Wörterbücher. Darüber hinaus bietet die TEI Komponenten sowohl für die Schemagenerierung als auch die Schema-Anpassung an und stellt integrierte Stylesheets zur Verfügung.

Author: ok, passt aus meiner Sicht

In vielen Fällen ist das Ziel von solchen ‘Prozeduren’ die Darstellung eines Dokuments in einer digitalen Druckvorlage. Wenn die Auszeichnung darauf abzielt, solche Darstellungsregeln zu definieren, spricht man von **"Präsentationalem Markup"**.

Je nachdem mit welchem Verständnis man an die Sache herangeht, sind die Grenzen natürlich fließend – letztendlich muss eine Software das Markup interpretieren und in eine Seitenbeschreibungssprache wie z.B. PDF übersetzen, die das Aussehen der Seite in absoluten Koordinaten beschreibt. Prinzipiell gibt XML nur die Regeln der Syntax an, in der Dokumente kodiert werden – nicht aber die Bedeutung der Auszeichnung. 

Bei der TEI geht es dezidiert darum, **Texte kodierend zu beschreiben**, daher ist sie vorwiegend **deskriptiv**.

Mit der TEI können unterschiedlichste **Aspekte eines Textes kodiert** werden. Einige relevante Fragen sind hier z.B.:

> - Wie ist der Text formal und logisch aufgebaut? Aus welchen Einheiten und Untereinheiten, Abschnitten, Gliederungsebenen etc. besteht er? Gibt es Überschriften, Zitate, Fußnoten, Anmerkungen usw.? Dies erfasst das **strukturelle Markup**.

Author: ok

> - Über welche ‘Dinge’ spricht der Text? Das können reale und fiktionale Entitäten sein (Personen, Orte, etc.), aber auch Konzepte, Ideen, andere Texte usw. Diese Aspekte werden im **semantischen Markup** annotiert.

Reviewer: Wie am 3.September besprochen, geht das in Ordnung. Siehe Review Meeting Dokument im Teams für DiTAH

- Wie spricht der Text? Mit welchen linguistischen, rhetorischen oder narratologischen Mitteln arbeitet er – Aspekte, die nicht an der Textoberfläche manifest sind, sondern erst durch einen Analyseschritt erfassbar werden, werden im **analytischen Markup** beschrieben.
- Und schließlich: Wie sieht der Text aus? Dies erfasst das **repräsentationale Markup**.

> Die folgenden Kapitel werden einen kurzen Überblick über die Struktur von TEI-Dokumenten geben und dabei einige sehr häufig verwendete Elemente hervorheben.

Reviewer 3: => Bitte diesen Satz ganz ans Ende stellen.

### Übung

<Quiz>
  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Zu welcher Form von Markup zählt TEI?**
      </Quiz.Question>

      <Quiz.MultipleChoice.Option isCorrect>
        **Prozedurales Markup** - TEI beschreibt Prozeduren, die ein Grafiker oder Setzer mit dem Text durchführen soll.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        **Deskriptives Markup** - TEI beschreibt unterschiedliche Aspekte der Texte selber, von Struktur, Semantik, über linguistische Aspekte bis hin zur Repräsentation des Textes.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>
        **Präsentationales Markup** - TEI definiert Darstellungsregeln für eine digitale Druckvorlage.
      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">

>         **Richtig!**

Reviewer: Ist mir nicht klar, warum nicht. Im Text sind Nachrichten für richtige und falsche Beantwortung hinterlegt.
@stefanprobst ?

>         **Richtig!**

Reviewer 4: works for me.

![Screenshot_20210903_151454](https://user-images.githubusercontent.com/20753323/132011026-68069dc9-1723-487e-944a-d4b886bf7ff8.png)

>         **Richtig!**

Reviewer 3: Richtig ist es nur, wenn nur "Deskriptives Markup" angeklickt ist. Weiter oben im Text steht:
_Möglich ist das, weil XML komplexe Inhalte beschreibt, ohne festzulegen, in welcher Form und mit welchen Computerprogrammen die Daten weiterverarbeitet werden sollen. Das bedeutet, es handelt sich um eine sogenannte 'Deskriptive' und keine 'Processing Markup Language'._

>         **Richtig!**

Reviewer 4: "Prozedurales Markup" war als richtige Antwort gekennzeichnet:

![Screenshot_20210903_161841](https://user-images.githubusercontent.com/20753323/132020535-3fe61a84-8899-4a57-a3e2-47e04350fdcb.png)

Hab das jetzt ausgebessert. In Zukunft bitte kurz selber checken before pinging me. thx

>         **Richtig!**

Reviewer 3: Ich habe es gleich im Vorschlagsmodus ausgebessert.

>         **Richtig!**

Reviewer 3: Ich habe es gleich im Vorschlagsmodus ausgebessert.

>         **Richtig!**

Reviewer 3: Liebe "Prinzessin", da wir im Moment direkt in Github arbeiten, kann es schon mal passieren, dass man was übersieht, weil es hier die Option etwas anzuklicken nicht gibt ....

>         **Richtig!**

Reviewer 4: dein ton ist völlig inakzeptabel.

>         **Richtig!**

Reviewer 3: Lieber Stefan,
eigenartig, dass gerade Du das sagst. Wenn man sich nicht auskennt, eine Frage hat, und man deshalb bei Kolleg*innen nachfragt und gesagt bekommt, dass man das **in Zukunft bitte selber checken soll**, das finde ich inakzeptabel. Teamarbeit zeichnet sich dadurch aus, dass man weiterhilft, selbst wenn mal jemand nur auf der "Leitung steht", und Teamarbeit bedingt halt auch, dass der Arbeitsalltag nicht völlig ungestört verläuft. Ich finde es unglaublich, sich so zu verhalten und zugleich über den Ton anderer aufzuregen, da hielte ich etwas Zurückhaltung für angebrachter!

>         **Richtig!**

Reviewer 5: Liebe Leute, 
bitte lieb sein zueinander!
Misstöne/Missverständnisse, bitte lieber persönlich/telefonisch versuchen zu klären, gerne unter Einbeziehung von mir oder Charly. Nicht hier.

---
title: "1. Markup-Sprachen: Von HTML zu TEI"
lang: de
date: 2021-07-23T10:36:13.905Z
version: 1.0.0
authors:
  - czeitschner-ulrike
  - schopper-daniel
editors:
  - königshofer-elisabeth
  - wiencek-florian
tags:
  - tei
  - xml
abstract: Dieses Kapitel gibt einen Überblick über den Digitalisierungsprozess und die Markup-Sprachen HTML und XML und stellt grundlegende Eigenschaften von TEI vor.
licence: ccby-4.0
toc: true
uuid: ycJztJbuEaDJyNFBehyEe
---
## Lerninhalte

- Überblick über den Digitalisierungsprozess: Vom physischen Original zum digitalen Dokument
- XML-Grundlagen: Elemente, Syntax, Dokumentenmodell und Hierarchie
- Wohlgeformtheit und Validität
- Schema und Namensraum
- Einordnung von TEI als fachspezifische Auszeichnungssprache für Textwissenschaften

## Vom analogen zum digitalen Text

> Sofern es sich nicht um ein 'born digital'-Dokument handelt, also ein Dokument, das mit einem Computer erstellt wurde und daher bereits in digitaler Form vorliegt, beginnt die

Reviewer 3: => Hier bitte die Absätze zusammenlegen.

> Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

Reviewer 3: => Hier bitte die Absätze zusammenlegen

XML stellt die Grundlage für zahlreiche **fachspezifische Auszeichnungssprachen** dar — bereits etablierte Beispiele dafür sind z.B. **GML**, die geografische Auszeichnungssprache für raumbezogene Informationen, **MathML**, die mathematische Auszeichnungssprache zur Beschreibung komplexer mathematischer Ausdrücke und Formeln, oder **MusicXML** für die westliche Notenschrift.

Auch in den Textwissenschaften ist XML heute die wichtigste Markup-Sprache, da es das ideale Werkzeug ist um Texte zu strukturieren, zu vermitteln und zu nutzen. Federführend in diesem Fachbereich ist die **Text Encoding Initiative**, kurz **TEI**. Doch bevor wir uns der TEI zuwenden, ist es notwendig, einige grundlegende Aspekte von XML näher zu betrachten.

### Quiz

<Quiz>
  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Welche der folgenden Aussagen sind richtig?** (mehrere Antworten möglich)
      </Quiz.Question>

      <Quiz.MultipleChoice.Option isCorrect>
        Durch die Verwendung von Markup-Sprachen werden die Strukturen von Texten für Maschinen lesbar.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>

>         HTML bildet die Grundlage für spezifische Markup-Sprachen wie TEI und MEI.

Reviewer 3: ```suggestion
        HTML bildet die Grundlage für spezifische Markup-Sprachen wie TEI und MusicXML.
```
=> Habe hier MEI mit MusicXML ersetzt, denn von MEI war nirgends die Rede.

      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>
        OCR sorgt dafür, dass die Textstruktur von Maschinen erkannt wird.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>

>         Cascading Style Sheets (CSS) und Java Script sind für die korrekte Darstellung von Inhalten notwendig

Reviewer 3: ```suggestion
        Cascading Style Sheets (CSS) und Java Script sind für die korrekte Darstellung von Inhalten notwendig.
```
=> Satzpunkt sollte einheitlich weg oder da sein, hier besser da.

### Übung

<Quiz>
  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Welche der folgenden Aussagen sind richtig?** (mehrere Antworten möglich)
      </Quiz.Question>

      <Quiz.MultipleChoice.Option isCorrect>
        TEI ist eine Auszeichnungssprache speziell für Texte.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        TEI basiert auf XML.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>
        XML stellt ein festes Vokabular für die Strukturierung von Dokumenten zur Verfügung.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        HTML ist eine Auszeichnungssprache, die vor allem auf die Strukturierung von Inhalten für die Veröffentlichung im World Wide Web entwickelt wurde.
      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">
        **Richtig!**

        TEI ist eine fachspezifische Markup-Sprache, die auf XML basiert und dazu dient, Dokumente für die textwissenschaftliche Forschung zu annotieren.

>         Andere Markup-Sprachen sind HTML und XML. Während HTML allerdings ein festes Vokabular anbietet, das für die Strukturierung von Inhalten für Webpublikationen ausgelegt ist, stellt XML als ein abstraktes Werzeug zur Modellierung und Strukturierung von Informationen lediglich Regeln bereit, wie Elemente zu definieren und anzuwenden sind. Man kann aber beliebig viele konkrete Auszeichnungssprachen mithilfe von XML definieren.

Reviewer 3: ```suggestion
        Andere Markup-Sprachen sind HTML und XML. Während HTML allerdings ein vordefiniertes Vokabular anbietet, das für die Strukturierung von Inhalten für Web-Publikationen ausgelegt ist, stellt XML als ein abstraktes Werkzeug zur Modellierung und Strukturierung von Informationen lediglich Regeln bereit, wie Elemente zu definieren und anzuwenden sind. Man kann mithilfe von XML beliebig viele konkrete Auszeichnungssprachen definieren.
```
=> Noch Fehler entdeckt ...

      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider nicht ganz richtig!**

        TEI ist eine fachspezifische Markup-Sprache, die auf XML basiert und dazu dient, Dokumente für die textwissenschaftliche Forschung zu annotieren.

>         Andere Markup-Sprachen sind HTML und XML. Während HTML allerdings ein festes Vokabular anbietet, das für die Strukturierung von Inhalten für Webpublikationen ausgelegt ist, stellt XML als ein abstraktes Werzeug zur Modellierung und Strukturierung von Informationen lediglich Regeln bereit, wie Elemente zu definieren und anzuwenden sind. Man kann aber beliebig viele konkrete Auszeichnungssprachen mithilfe von XML definieren.

Reviewer 3: ```suggestion
        Andere Markup-Sprachen sind HTML und XML. Während HTML allerdings ein vordefiniertes Vokabular anbietet, das für die Strukturierung von Inhalten für Web-Publikationen ausgelegt ist, stellt XML als ein abstraktes Werkzeug zur Modellierung und Strukturierung von Informationen lediglich Regeln bereit, wie Elemente zu definieren und anzuwenden sind. Man kann aber beliebig viele konkrete Auszeichnungssprachen mithilfe von XML definieren.
```
=> Noch Fehler entdeckt ...

```xml
a) <element attribut="wert">Inhalt</element>
b) <element attribut="wert" attribut2="wert"/>
```

### Wohlgeformtheit

Jede XML-Annotation bedient sich dieser einfachen Bausteine. In der konkreten Anwendung sind jedoch besondere Regeln einzuhalten. U.a. syntaktische Vorschriften.

Die wichtigste davon ist die Einhaltung der **Wohlgeformtheit** oder **well-formedness**. Sie bedeutet, dass **jedes geöffnete Element auch wieder geschlossen** werden muss, wie im untenstehenden Beispiel die Elemente a und b. **Andernfalls** kann es sich nur um ein **Empty-Element** handeln, wie z.B. das im Beispiel angeführte leere c-Element. 

```xml
<a>Inhalt</a> Inhalt <b>Inhalt</b> Inhalt ... <c/>
```

Sollten **Elemente verschachtelt** sein, so müssen sie **in der richtigen Reihenfolge geöffnet und geschlossen werden**. Dies gilt auch für den Fall, dass mehrere Phänomene einander überlappen. 

_Falsch:_

```xml
<a>Inhalt <b>Inhalt</a> Inhalt</b>
```

_Richtig:_

```xml
<a>Inhalt <b>Inhalt</b></a> <b>Inhalt</b>
```

<SideNote type="note" title="Merke">

> Die Regel der "well-formedness" ist zwingend, d.h. **alle XML-Dokumente müssen wohlgeformt sein!**

Reviewer 3: ```suggestion
Die Regel der 'well-formedness' ist zwingend, d.h. **alle XML-Dokumente müssen wohlgeformt sein!**
```

        XML besitzt keinerlei Regelwerk. Wohlgeformtheit und Validität sind lediglich Empfehlungen.
      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">
        **Richtig!**

        Die Wohlgeformtheit eines Dokuments ist zwingend notwendig. Es wird aber empfohlen ein Dokument ebenso valide zu gestalten.
      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider falsch.**

        Die Wohlgeformtheit eines Dokuments ist zwingend notwendig. Es wird aber empfohlen ein Dokument ebenso valide zu gestalten.
      </Quiz.Message>
    </Quiz.MultipleChoice>
  </Quiz.Card>
</Quiz>

### Elementhierarchie in XML

Ein XML-Dokument ist eine hierarchische Strukturierung und Anordnung von Inhalten. Diese ist jedoch nicht beliebig, sondern wird im Dokument-Schema festgelegt. Und da Dokumente gültig sein sollen, kann man sich die Strukturierung nicht willkürlich aussuchen.

**Das Schema schreibt vor, welche Elemente es gibt, wo sie anzutreffen sein dürfen und wo nicht.** So kann beispielsweise im Schema definiert werden, dass ein Element nur ein bestimmtes Elternelement haben kann.

- Ein Vorname darf nur das Kind eines Personennamens sein.

> - Ein Personenname kann nur _innerhalb_ einer Überschrift, eines Paragraphen oder einer Verszeile vorkommen, aber niemals außerhalb.

Reviewer 3: ```suggestion
- Oder darf ein Personenname nur _innerhalb_ einer Überschrift, eines Paragraphen oder einer Verszeile vorkommen, jedoch niemals außerhalb.
```
=> Irre ich mich, oder wurde auch hier erneut umgeschrieben und gleich committed? Bitte bei Einwänden gegen Vorschläge nachvollziehbare Kommentare einfügen. Ich ändere hier wieder um, weil die Lesbarkeit auf den einführenden Satz vor dem Doppelpunkt abgestimmt sein soll.

Reviewer 2: Umgeschrieben wurde nicht, es kann aber sein, dass ich ein früheres comment eingearbeitet habe und es jetzt nicht mehr passt - ich bin mir auch nicht sicher, dass jetzt die Sätze passen, wenn ich es mit dem Beginn "oder darf..." übernehmen... kann das wer überprüfen?
Und mir ist aufgefallen: ph oder f? (Kiki du hast nach Paragraf geändert; Elisabeth und Florian haben öfters Paragraph gehabt - das sollten wir am Ende nochmal vereinheitlichen bzw. prüfen!

Reviewer 3: => Völlig richtig, Danke fürs aufmerksame Lesen! Da hat ein "es" vor "darf" gefehlt - jetzt sollte es passen.
=> ""ph" habe ich bereits überall zu "f" ausgebessert ... richte mich nach den Angaben im Duden ... habe hoffentlich nichts übersehen.

Im Wurzelelement befindet sich ein Attribut mit der Bezeichnung **namespace**, das als Wert einen Unified Ressource Identifier (URI) enthält, der angibt, woher die verwendeten Elemente und Attribute stammen. 

Handelt es sich z.B. um ein **Dokument, das nach den Richtlinien der TEI** erstellt wurde, ist der **Name des Root-Elements TEI**. Der Attributwert der namespace declaration verweist darauf, dass ausschließlich von der TEI definierte Elemente und Attribute verwendet wurden. 

![TEI Namensraum](images/tei-namensraum.png "TEI-Namensraum. Bild: Ulrike Czeitschner, ACDH-CH")

Möchte man Komponenten aus verschiedenen Namensräumen mischen, müssen zwei Namenraum-Deklarationen angegeben werden, die jeweils unterschiedliche Präfixe definieren, z.B. ein Attribut xmlns:tei und ein Attribut xmlns:xyz. 

Im Dokument selbst müssen diese Präfixe, gefolgt von einem Doppelpunkt, den betreffenden Elementnamen vorangestellt werden um deren Schema-Zugehörigkeit anzuzeigen:

![mehrere Namensraeume](images/mehrerenamensraeumexml.png "Mehrere Namensräume. Bild: Ulrike Czeitschner, ACDH-CH")

<SideNote type="note" title="Hinweis">

> Kommentare sind außerhalb von einführenden oder schließenden Tags überall erlaubt. Sie werden eingeleitet von einer Spitzklammer, einem Rufzeichen, sowie zwei Minuszeichen. Dann folgt der Kommentartext und danach zwei Minuszeichen und eine schließende Spitzklammer:

Reviewer 3: ```suggestion
Kommentare sind außerhalb von einführenden oder schließenden Tags überall erlaubt. Sie werden eingeleitet von einer Spitzklammer, einem Rufzeichen, sowie zwei Minuszeichen. Dann folgt der Kommentartext, danach zwei Minuszeichen und eine schließende Spitzklammer:
```

- Wie ist der Text formal und logisch aufgebaut? Aus welchen Einheiten und Untereinheiten, Abschnitten, Gliederungsebenen etc. besteht er? Gibt es Überschriften, Zitate, Fußnoten, Anmerkungen usw.? Dies erfasst das **strukturelle Markup**.
- Über welche ‘Dinge’ spricht der Text? Das können reale und fiktionale Entitäten sein (Personen, Orte, etc.), aber auch Konzepte, Ideen, andere Texte usw. Diese Aspekte werden im **semantischen Markup** annotiert.
- Wie spricht der Text? Mit welchen linguistischen, rhetorischen oder narratologischen Mitteln arbeitet er – Aspekte, die nicht an der Textoberfläche manifest sind, sondern erst durch einen Analyseschritt erfassbar werden, werden im **analytischen Markup** beschrieben.
- Und schließlich: Wie sieht der Text aus? Dies erfasst das **repräsentationale Markup**.

Die folgenden Kapitel werden einen kurzen Überblick über die Struktur von TEI-Dokumenten geben und dabei einige sehr häufig verwendete Elemente hervorheben.

### Übung

<Quiz>
  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Zu welcher Art von Markup zählt TEI?**
      </Quiz.Question>

>       <Quiz.MultipleChoice.Option isCorrect>

Reviewer 3: ```suggestion
      <Quiz.MultipleChoice.Option>
```

        **Prozedurales Markup** – TEI beschreibt Prozeduren, die ein Setzer oder Grafiker mit dem Text durchführen soll.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>
        **Deskriptives Markup** - TEI beschreibt unterschiedliche Aspekte der Texte selber, von Struktur, Semantik, über linguistische Aspekte bis hin zur Repräsentation des Textes.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option>

>         **Präsentationales Markup** – TEI definiert Darstellungsregeln für eine digitale Druckvorlage.

Reviewer 3: ```suggestion
        **Repräsentationales Markup** – TEI definiert Darstellungsregeln für eine digitale Druckvorlage.
```
=> Ist mir erst jetzt aufgefallen: Diese Frage ist beinahe gemein, denn TEI enthält auch Repräsentationales = renditional markup, definiert aber keine Darstellungsregeln ...

Author: super, ja, das ist auf jeden Fall ein wichtiger Punkt!

      </Quiz.MultipleChoice.Option>

      <Quiz.Message type="correct">
        **Richtig!**

        TEI ist vorwiegend deskriptiv.
      </Quiz.Message>

      <Quiz.Message type="incorrect">
        **Leider falsch.**

>         TEI zählt zum deskriptiven Markup. Vom logischen und formalen Aufbau eines Textes, seiner Semantik bis hin zu seiner Repräsentation können unterschiedlichste Aspekte kodiert werden.

Reviewer 3: ```suggestion
        TEI zählt zum deskriptiven Markup. Vom logischen und formalen Aufbau eines Textes, seiner Semantik bis hin zu seiner Repräsentation können unterschiedlichste Aspekte kodiert werden. Zwar werden TEI-Elemente auch für die Textrepräsentation genutzt, allerdings definiert TEI keine konkreten Darstellungsregeln.
```

---
title: "1. Markup-Sprachen: Von HTML zu TEI"
lang: de
date: 2021-07-23T10:36:13.905Z
version: 1.0.0
authors:
  - czeitschner-ulrike
  - schopper-daniel
editors:
  - königshofer-elisabeth
  - wiencek-florian
tags:
  - tei
  - xml
abstract: Dieses Kapitel gibt einen Überblick über den Digitalisierungsprozess
  und die Markup-Sprachen HTML und XML und stellt grundlegende Eigenschaften von
  TEI vor.
licence: ccby-4.0
toc: true
uuid: ycJztJbuEaDJyNFBehyEe
---
## Lerninhalte

- Überblick über den Digitalisierungsprozess: Vom physischen Original zum digitalen Dokument
- XML-Grundlagen: Elemente, Syntax, Dokumentenmodell und Hierarchie
- Wohlgeformtheit und Validität
- Schema und Namensraum
- TEI – Markup-Sprache und Dokumentenformat für die Textwissenschaften

## Vom analogen zum digitalen Text

> Sofern es sich nicht um ein 'born digital'-Dokument handelt, also ein Dokument, das mit einem Computer erstellt wurde und daher bereits in digitaler Form vorliegt, beginnt die

Reviewer 3: ```suggestion
Sofern es sich nicht um ein 'born digital'-Dokument handelt, also ein Dokument, das mit einem Computer erzeugt wurde und daher bereits in digitaler Form vorliegt, beginnt die
```

Erstellung einer digitalen Text-Ressource mit dem **[Einscannen einer Vorlage](https://dhmuseum.uni-trier.de/node/5)**. Dieser Scan resultiert in einem Pixelbild  - z.B. im TIF, JPG oder PNG-Format - oder einem PDF. Das Aussehen der Print-Quelle ist in den Bilddaten ersichtlich, aber der textuelle Inhalt ist noch nicht digital weiter verarbeitbar. 

> In einem zweiten Schritt erfolgt daher die **Texterkennung** oder **['Optical Character Recognition' (OCR)](https://de.wikipedia.org/wiki/Texterkennung)**. Das Ergebnis ist ein elektronischer Text, z.B. in Form einer Text-Datei oder eines Word-Dokuments, den man zwar mehr oder weniger strukturieren und formatieren kann, wobei die Möglichkeiten sowohl im Hinblick auf das Layout als auch die inhaltliche Beschreibung häufig sehr begrenzt sind. Außerdem ist die Konvertierung in andere Formate oft nicht verlustfrei möglich, wenn die Formatierungs- oder Strukturinformationen in programmspezifischen Formaten abgespeichert sind. ([\> mehr Informationen zur Volltextdigitalisierung](https://dhmuseum.uni-trier.de/node/49))

Reviewer 3: ```suggestion
In einem zweiten Schritt erfolgt daher die **Texterkennung** oder **['Optical Character Recognition' (OCR)](https://de.wikipedia.org/wiki/Texterkennung)**. Das Ergebnis ist ein elektronischer Text, z.B. in Form einer Text-Datei oder eines Word-Dokuments, den man zwar mehr oder weniger strukturieren und formatieren kann, wobei die Möglichkeiten sowohl im Hinblick auf das Layout als auch die inhaltliche Beschreibung häufig sehr begrenzt sind. Außerdem ist die Konvertierung in andere Formate oft nicht verlustfrei möglich, wenn die Formatierungs- oder Strukturinformationen in programmspezifischen Formaten abgespeichert sind. ([\> Mehr Informationen zur Volltextdigitalisierung](https://dhmuseum.uni-trier.de/node/49))
```

Das Konsortium besteht aus einem fixen Team mit mehr als 60 Personen. Die vier Gastgeberorganisationen sind das MIT Computer Science and Artificial Intelligence Laboratory (MIT/CSAIL) in den USA, das European Research Consortium for Informatics and Mathematics (ERCIM) in Frankreich, die KEIO University in Japan und die Beihang University in China. Dazu kommen über 400 Mitgliederorganisationen aus mehr als 30 Ländern. Ergänzt wird die Arbeit dieser beiden Gruppen durch Beiträge aus der interessierten Web-Community, die an den Entwicklungs- und Diskussionsprozessen maßgeblich beteiligt ist. Ihre Mitarbeit ist ein Garant dafür, dass die Empfehlungen nicht nur sehr praxisorientiert und erprobt sind, sondern den Nutzer*innen z.B. Übersetzungen der Empfehlungen in über 40 Sprachen und einzelne Implementierungen als Open Source zur Verfügung stehen.

> Erstmals als Empfehlung veröffentlicht wurde XML 1998, die derzeit aktuelle [5. Ausgabe](w3.org/TR/xml/) stammt vom November 2008.

Reviewer 3: ```suggestion
Erstmals als Empfehlung veröffentlicht wurde XML 1998, die derzeit aktuelle [5. Ausgabe](https://www.w3.org/TR/xml/) stammt vom November 2008.
```
=> Link funktionierte nicht.

> ### Übung

Reviewer 3: ```suggestion
### Quiz
```

<Quiz>
  <Quiz.Card>
    <Quiz.MultipleChoice>
      <Quiz.Question>
        **Welche der folgenden Aussagen sind richtig?** (mehrere Antworten möglich)
      </Quiz.Question>

      <Quiz.MultipleChoice.Option isCorrect>
        TEI ist eine Auszeichnungssprache speziell für Texte.
      </Quiz.MultipleChoice.Option>

      <Quiz.MultipleChoice.Option isCorrect>

>         TEI basiert auf XML.

Reviewer 3: => Dass TEI auf XML basiert, erfährt man erst viel weiter unten ... entweder Frage hier rausnehmen oder Quiz verschieben???

```xml
a) <element></element>
b) <element/>
```

Zur **Beschreibung komplexer Inhalte** reicht ein Elementname alleine meist nicht aus, weshalb mehrere verschiedene Elemente für denselben Inhalt vergeben werden können. Zusatzinformationen lassen sich jedoch auch unterbringen, indem man im Start-Tag eines Elements sogenannte **Attribute** einfügt, die umgeben von Anführungszeichen bestimmte **Werte** enthalten. Ein Element darf selbstverständlich mehrere unterschiedliche Attribute besitzen. Grundsätzlich können Element-, Attribut- und Wertebezeichnungen aus jeder beliebigen Sprache gewählt werden, im Sinne der Austauschbarkeit haben sich jedoch englische Bezeichnungen durchgesetzt.

Es gibt also a) **Tags**, die zusammen mit den Inhalten **Elemente** bilden und b) **leere Elemente**, die mit Hilfe von **Attributen** und **Attributwerten** genauer spezifiziert werden können:

```xml
a) <element attribut="wert">Inhalt</element>
b) <element attribut="wert" attribut2="wert"/>
```

## XML-Syntax

Jede XML-Annotation bedient sich dieser einfachen Bausteine. In der konkreten Anwendung sind jedoch besondere Regeln einzuhalten. U.a. syntaktische Vorschriften.

### Wohlgeformtheit

> Die wichtigste Syntaxregel ist die Einhaltung der **Wohlgeformtheit** oder **well-formedness**. Sie bedeutet, dass **jedes geöffnete Element auch wieder geschlossen** werden muss, wie die hier angegebenen Elemente a und b. **Andernfalls** kann es sich nur um ein **leeres Element** handeln, wie das angeführte c-Element: 

Reviewer 3: ```suggestion
Die wichtigste Syntaxregel ist die Einhaltung der **Wohlgeformtheit** oder **'well-formedness'**. Sie bedeutet, dass **jedes geöffnete Element auch wieder geschlossen** werden muss, wie die hier angegebenen Elemente a und b. **Andernfalls** kann es sich nur um ein **leeres Element** handeln, wie das angeführte c-Element: 
```

```xml
<a>Inhalt</a> Inhalt <b>Inhalt</b> Inhalt ... <c/>
```

Sollten Elemente **verschachtelt** sein, so müssen sie in der **richtigen Reihenfolge geöffnet und geschlossen** werden. Dies gilt auch für den Fall, dass sich mehrere Elemente überlappen: 

_Falsch:_

```xml
<a>Inhalt <b>Inhalt</a> Inhalt</b>
```

_Richtig:_

```xml
<a>Inhalt <b>Inhalt</b></a> <b>Inhalt</b>
```

<SideNote type="note" title="Merke">
Die Regel der 'well-formedness' ist zwingend, d.h. **alle XML-Dokumente müssen wohlgeformt sein!**

- Geöffnete Elemente müssen wieder geschlossen werden oder leere Elemente sein. 
- Bei verschachtelten bzw. sich überlappenden Elementen ist die richtige Reihenfolge beim Öffnen und Schließen von Elementen zu berücksichtigen.
</SideNote>

### Validität

Darüber hinaus kann ein XML-Dokument auch valide oder gültig sein. Was bedeutet das? 

> Wie schon erwähnt, kann im Grunde jeder mit XML seine eigene Markup-Sprache für bestimmte Zwecke definieren. Gleichzeitig ist es mit Blick auf kollaborative Arbeitsweisen und Open Science unerlässlich, die erzeugten Dokumente weitergeben und mehrfach nachnutzen zu können. **Wer also eine Sprache "erfindet", sollte nicht nur das Vokabular, sprich die Elemente, Attribute und Werte festlegen, sondern auch eine "Grammatik" – Schema genannt – schreiben**, in der definiert ist, was die Elemente bedeuten, in welcher Beziehung sie zueinander stehen, wo und wie sie verwendet werden dürfen. 

Reviewer 3: ```suggestion
Wie schon erwähnt, kann im Grunde jede\*r mit XML seine eigene Markup-Sprache für bestimmte Zwecke definieren. Gleichzeitig ist es mit Blick auf kollaborative Arbeitsweisen und Open Science unerlässlich, die erzeugten Dokumente weitergeben und mehrfach nachnutzen zu können. **Wer also eine Sprache "erfindet", sollte nicht nur das Vokabular, sprich die Elemente, Attribute und Werte festlegen, sondern auch eine "Grammatik" – Schema genannt – schreiben**, in der definiert ist, was die Elemente bedeuten, in welcher Beziehung sie zueinander stehen, wo und wie sie verwendet werden dürfen. 
```

- Eindeutige ID-Nummern dürfen sinnvollerweise nur einmal vergeben werden. 
- Ebenso sollte in einem Geburtsregister jede Person nur ein Geburtsdatum haben.

Die Struktur eines Buches könnte in XML etwa wie folgt abgebildet werden: 

![XML Hierarchie eines Buches](images/xml-hierarchie_neu.png "XML-Struktur eines Buches. Bild: Ulrike Czeitschner, ACDH-CH")

Innerhalb des Wurzelelements befindet sich ein div-Element ("div" steht für 'division', d.h. Textabschnitt) mit dem Attribut `@type=book` für das Buch, das die Kindelemente Vorwort und mehrere Kapitel enthält, vielleicht gefolgt von einem Nachwort, eventuell einem Register und einem Impressum. Diese Kindelemente verhalten sich zueinander alle wie Geschwister und sind selbst Eltern der Titel-, Paragrafen- und Verszeilen-Elemente, die ebenfalls Geschwister sind und selbstverständlich auch wieder Kinder haben können. Die so strukturierten Inhalte spiegeln den Aufbau des Buches wider.

### Aufbau von XML-Dokumenten

![Aufbau XML Dokument](images/xml-dokument_neu.png "Aufbau eines XML-Dokuments. Bild: Ulrike Czeitschner, ACDH-CH")

> Die obenstehende Grafik zeigt den Aufbau eines XML-Dokuments. Jedes XML-Dokument beginnt mit einem **Prolog**, der zumindest die **XML-Deklaration** ('XML declaration') enthält, die angibt, um welche XML-Version es sich handelt. Die Tags des Prologs zeichnen sich durch eine besondere Schreibweise aus: sie sind einteilig und an den Fragezeichen vorne und hinten zu erkennen. 

Reviewer 3: ```suggestion
Die hier angeführte Abbildung zeigt den Aufbau eines XML-Dokuments. Jedes XML-Dokument beginnt mit einem **Prolog**, der zumindest die **XML-Deklaration** ('XML declaration') enthält, die angibt, um welche XML-Version es sich handelt. Die Tags des Prologs zeichnen sich durch eine besondere Schreibweise aus: sie sind einteilig und an den Fragezeichen vorne und hinten zu erkennen. 
```

Im Wurzelelement befindet sich ein Attribut mit der Bezeichnung **Namensraum** ('namespace') `@xmlns`, das als Wert einen 'Unified Ressource Identifier' (URI) enthält, der angibt, woher die verwendeten Elemente und Attribute stammen. 

Handelt es sich z.B. um ein **Dokument**, das **nach den Richtlinien der TEI** erstellt wurde, ist der **Name des Wurzelelements TEI**. Der Attributwert von `@xmlns` verweist darauf, dass ausschließlich von der TEI definierte Elemente und Attribute verwendet wurden. 

![TEI Namensraum](images/tei-namensraum_neu.png "TEI-Namensraum. Bild: Ulrike Czeitschner, ACDH-CH")

Möchte man Komponenten aus verschiedenen Namensräumen mischen, müssen zwei Namensraum-Deklarationen angegeben werden, die jeweils unterschiedliche Präfixe definieren, z.B. ein Attribut `@xmlns:tei` und ein Attribut `@xmlns:xyz`. 

Im Dokument selbst müssen diese Präfixe, gefolgt von einem Doppelpunkt, den betreffenden Elementnamen vorangestellt werden um deren Schema-Zugehörigkeit anzuzeigen:

![mehrere Namensraeume](images/mehrerenamensraeumexml_neu.png "Mehrere Namensräume. Bild: Ulrike Czeitschner, ACDH-CH")

> <SideNote type="note" title="Hinweis">

Reviewer 3: => "Hinweis" oder "Merke" oder eh beides? Wenn beides was wofür genau?

Author: Das sollten wir uns noch mal genau ansehen. Intuitiv würde ich Merke vor allem für Merksätze verwenden / wichtige Regeln. Hinweis sind weniger "dogmatisch" und mehr eine Hervorhebung. Man könnte auch überlegen, dem Hinweis ein anderes Icon zu geben...

      <Quiz.Message type="incorrect">
        **Leider nicht ganz richtig.**

        Die XML-Deklaration, der Schema-Verweis und der Verarbeitungsverweis können Teil des Prologs einer XML-Datei sein.

        Die Namensraum-Deklaration wird im Wurzelelement angegeben, das unmittelbar auf den Prolog folgt.
      </Quiz.Message>
    </Quiz.MultipleChoice>
  </Quiz.Card>
</Quiz>

## Von XML zu TEI

Das TEI-Dokumentenformat, herausgegeben von der 1987 gegründeten **[Text Encoding Initiative](https://tei-c.org/)**, basiert auf XML und ist mittlerweile der de-facto-Standard für die Kodierung von Texten und sprachlichen Informationen. Die derzeit empfohlene Version P5 wurde 2007 veröffentlicht und seitdem regelmäßig verbessert und aktualisiert. Das Regelwerk enthält mehr als 500 Elemente und ebenso zahlreiche Attribute, die in sachbezogenen [Modulen](https://tei-c.org/release/doc/tei-p5-doc/en/html/index.html) organisiert sind. Sie berücksichtigen Lyrik und Dramentexte ebenso wie Handschriften, kritische Apparate oder Wörterbücher. Darüber hinaus bietet die TEI Komponenten sowohl für die Schema-Generierung als auch -Anpassung an und stellt integrierte Stylesheets zur Verfügung.

> Der Begriff "Markup" stammt ursprünglich aus der Druckersprache und bedeutet "Textauszeichung". Ein Beispiel dafür wären "Fahnenkorrekturen". Sie enthalten z.B. Fehlerberichtigungen und Anweisungen, was mit den angestrichenen Stellen zu tun ist – Buchstabenverdreher umstellen, Doppelungen löschen, Absätze aufteilen usw. Dabei werden Prozeduren vorgegeben, die früher der Setzer, heute der Grafiker mit dem Text zu vollziehen hat. Das könnte man als **"Prozedurales Markup"** bezeichnen.

Reviewer 3: ```suggestion
Der Begriff "Markup" stammt ursprünglich aus der Druckersprache und bedeutet "Textauszeichung". Ein Beispiel dafür wären Fahnenkorrekturen. Sie enthalten z.B. Fehlerberichtigungen und Anweisungen, was mit den angestrichenen Stellen zu tun ist – Buchstabenverdreher umstellen, Doppelungen löschen, Absätze aufteilen usw. Dabei werden Prozeduren vorgegeben, die früher der Setzer, heute der Grafiker mit dem Text zu vollziehen hat. Das könnte man als **"Prozedurales Markup"** bezeichnen.
```

In vielen Fällen ist das Ziel von solchen "Prozeduren" die Darstellung eines Dokuments in einer digitalen Druckvorlage. Wenn die Auszeichnung darauf abzielt, solche Darstellungsregeln zu definieren, spricht man vom **"Präsentationalen Markup"**.

Je nachdem mit welchem Verständnis man an die Sache herangeht, sind die Grenzen natürlich fließend – letztendlich muss eine Software das Markup interpretieren und in eine Seitenbeschreibungssprache wie z.B. PDF übersetzen, die das Aussehen der Seite in absoluten Koordinaten beschreibt. XML gibt prinzipiell nur die Regeln der Syntax an, in der Dokumente kodiert werden – nicht aber die Bedeutung der Auszeichnungselemente. 

Im Gegensatz dazu, geht es im TEI-Dokumentenformat dezidiert darum, **Texte kodierend zu beschreiben**. Da es vorwiegend deskriptiv ist, kann man TEI auch als **Deskriptives Markup** bezeichnen.

Mit TEI lassen sich unterschiedlichste **Aspekte von Texten kodieren**. Relevante Fragen in diesem Zusammenhang wären z.B.:

> - Wie ist der Text formal und logisch aufgebaut? Aus welchen Einheiten und Untereinheiten, Abschnitten, Gliederungsebenen etc. besteht er? Gibt es Überschriften, Zitate, Fußnoten, Anmerkungen usw.? > **Strukturelles Markup**

Reviewer 3: ```suggestion
- Wie ist der Text formal und logisch aufgebaut? Aus welchen Einheiten und Untereinheiten, Abschnitten, Gliederungsebenen etc. besteht er? Gibt es Überschriften, Zitate, Fußnoten, Anmerkungen usw.? **> Strukturelles Markup**
```
=> Habe Spitzklammer auch fett gemacht - wie vorne beim Verweis auf die Uni Trier.

> - Über welche "Dinge" spricht der Text? Das können reale und fiktionale Entitäten sein (Personen, Orte etc.), aber auch Konzepte, Ideen, andere Texte usw. > **Semantisches Markup**

Reviewer 3: ```suggestion
- Über welche "Dinge" spricht der Text? Das können reale und fiktionale Entitäten sein (Personen, Orte etc.), aber auch Konzepte, Ideen, andere Texte usw. **> Semantisches Markup**
```

> - Wie spricht der Text? Mit welchen linguistischen, rhetorischen oder narratologischen Mitteln arbeitet er – Aspekte, die nicht an der Textoberfläche manifest sind, sondern erst durch Analyseschritte erfassbar werden. > **Analytisches Markup**

Reviewer 3: ```suggestion
- Wie spricht der Text? Mit welchen linguistischen, rhetorischen oder narratologischen Mitteln arbeitet er – Aspekte, die nicht an der Textoberfläche manifest sind, sondern erst durch Analyseschritte erfassbar werden. **> Analytisches Markup**
```

> - Und schließlich: Wie sieht der Text aus? > **Repräsentationales Markup**

Reviewer 3: ```suggestion
- Und schließlich: Wie sieht der Text aus? **> Repräsentationales Markup**
```

Um elektronische Daten systematisch, am besten mehrfach und interdisziplinär für die sprach-, literatur- oder kulturwissenschaftliche Forschung nutzen zu können, benötigen sie **dauerhafte**, strukturelle, semantische und linguistische **Annotationen**. An diesem Punkt kommen sogenannte Auszeichnungs- oder Markup-Sprachen zum Einsatz. 

## Markup-Sprachen

Auszeichnungssprachen dienen der Beschreibung von unterschiedlichsten Daten z.B. von Texten, Bildern, Filmen usw. Gleichzeitig dienen sie auch der Beschreibung des Verfahrens, das zu ihrer Weiterverarbeitung bzw. den Datenaustausch notwendig ist. **Markup-Sprachen** sind Beschreibungssprachen, die es erlauben, die Gliederung und Formatierung, aber auch inhaltliche Aspekte einer Vorlage wie z.B. Texte in maschinenlesbarer Form zu annotieren. 

> Die bekannteste Markup-Sprache ist die ['Hypertext Markup Language' (HTML)](https://www.w3.org/html/), die von Tim Berners-Lee entwickelt und vom [World Wide Web Consortium (W3C)](https://www.w3.org/) als Web-Standard veröffentlicht wurde. Sie stellt eine der technologischen Grundlagen des World Wide Web dar und gliedert die multimedialen Inhalte einer Webseite so, dass sie von Webbrowsern interpretiert und dargestellt und von Suchmaschinen gefunden werden können. Für die Gliederung von z.B. Überschriften, Absätzen, Zitaten, Listen, Tabellen, Hyperlinks etc. steht ein beschränktes Vokabular an vordefinierten Auszeichnungselementen, auch 'Tags' genannt, zur Verfügung. HTML übernimmt dabei die Strukturierung der Inhalte, während andere Technologien wie [Cascading Stylesheets (CSS)](https://www.w3.org/Style/CSS/Overview.en.html) für die visuelle Darstellung oder [JavaScript](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/) für die Integration von dynamischen, interaktiven Elementen zum Einsatz kommen.

Reviewer 3: ```suggestion
Die bekannteste Markup-Sprache ist die ['Hypertext Markup Language' (HTML)](https://www.w3.org/html/), die von Tim Berners-Lee entwickelt und vom ['World Wide Web Consortium' (W3C)](https://www.w3.org/) als Web-Standard veröffentlicht wurde. Sie stellt eine der technologischen Grundlagen des World Wide Web dar und gliedert die multimedialen Inhalte einer Webseite so, dass sie von Webbrowsern interpretiert und dargestellt und von Suchmaschinen gefunden werden können. Für die Gliederung von z.B. Überschriften, Absätzen, Zitaten, Listen, Tabellen, Hyperlinks etc. steht ein beschränktes Vokabular an vordefinierten Auszeichnungselementen, auch 'Tags' genannt, zur Verfügung. HTML übernimmt dabei die Strukturierung der Inhalte, während andere Technologien wie ['Cascading Stylesheets' (CSS)](https://www.w3.org/Style/CSS/Overview.en.html) für die visuelle Darstellung oder ['JavaScript'](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/) für die Integration von dynamischen, interaktiven Elementen zum Einsatz kommen.
```

Für die wissenschaftliche Nutzung von Daten bietet HTML mit seinem beschränkten und auf die Darstellung von Web-Inhalten fokussierten Vokabular geringe Flexibilität in der Auszeichnung und schränkt damit sowohl die Datenverarbeitung als auch die -auswertung ein. Im Gegensatz dazu, ist die ['Extensible Markup Language' (XML)](https://www.w3.org/XML/) oder "erweiterbare Auszeichnungssprache" eher ein abstraktes Werkzeug zur Modellierung und Strukturierung von Informationen und zur Haltung von Daten mit potentiell unbegrenzten Möglichkeiten.

XML stellt die Grundlage für zahlreiche **fachspezifische Auszeichnungssprachen** dar — bereits etablierte Beispiele dafür sind z.B. **GML**, die geografische Auszeichnungssprache für raumbezogene Informationen, **MathML**, die mathematische Auszeichnungssprache zur Beschreibung komplexer mathematischer Ausdrücke und Formeln, oder **MusicXML** für die westliche Notenschrift.

> Auch in den Textwissenschaften ist XML heute die wichtigste Markup-Sprache, da es das ideale Werkzeug ist um Texte zu strukturieren, zu vermitteln und zu nutzen. Federführend in diesem Fachbereich ist die **Text Encoding Initiative**, kurz **TEI**. Doch bevor wir uns der TEI zuwenden, ist es notwendig, einige grundlegende Aspekte von XML näher zu betrachten.

Reviewer 3: ```suggestion
Auch in den Textwissenschaften ist XML heute die wichtigste Markup-Sprache, da es das ideale Werkzeug ist um Texte zu strukturieren, zu vermitteln und zu nutzen. Federführend in diesem Fachbereich ist die **'Text Encoding Initiative'**, kurz **TEI**. Doch bevor wir uns der TEI zuwenden, ist es notwendig, einige grundlegende Aspekte von XML näher zu betrachten.
```
=> Nur fehlende einfache Anführungszeichen ergänzt.

Die Bezeichnung von XML als "Markup Language" ist leider nicht sehr präzise gewählt, wenn nicht sogar irreführend. Denn XML definiert keine Elemente. Es stellt vielmehr ein Regelwerk für den Entwurf von konkreten Auszeichnungssprachen und ihren Anwendungen dar und wird deshalb auch als Metasprache bezeichnet, die unbegrenzt erweiterbar ist. Das bedeutet, dass sich mit XML beliebig viele unterschiedliche Markup-Sprachen definieren lassen, die jeweils ganz unterschiedliche Zwecke verfolgen können. XML selbst ist leicht verständlich, da es nicht aus abstrakten Codes, sondern aus Text besteht, was den Vorteil hat, dass diese Sprache von Mensch und Maschine gelesen werden kann. Dies ist neben der Plattformunabhängigkeit und der Erweiterbarkeit mit ein Grund für die weite Verbreitung und den Erfolg von XML.

> Verantwortlich für die Entwicklung von XML ist das 1994 gegründete [World Wide Web Consortium](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.

Reviewer 3: ```suggestion
Verantwortlich für die Entwicklung von XML ist das 1994 gegründete ['World Wide Web Consortium'](https://www.w3.org/) unter der Leitung von Tim Berners-Lee, dem Erfinder der Hypertext Markup Language und des World Wide Web. Dieses internationale Gremium bemüht sich darum, Web-Standards zu entwickeln und Richtlinien zu den einzelnen Spezifikationen herauszugeben mit dem Ziel, das Potential und die Reichweite des Web zu vergrößern, es langfristig zu sichern und von einzelnen Herstellern unabhängig zu erhalten.
```

> Das Konsortium besteht aus einem fixen Team mit mehr als 60 Personen. Die vier Gastgeberorganisationen sind das MIT Computer Science and Artificial Intelligence Laboratory (MIT/CSAIL) in den USA, das European Research Consortium for Informatics and Mathematics (ERCIM) in Frankreich, die KEIO University in Japan und die Beihang University in China. Dazu kommen über 400 Mitgliederorganisationen aus mehr als 30 Ländern. Ergänzt wird die Arbeit dieser beiden Gruppen durch Beiträge aus der interessierten Web-Community, die an den Entwicklungs- und Diskussionsprozessen maßgeblich beteiligt ist. Ihre Mitarbeit ist ein Garant dafür, dass die Empfehlungen nicht nur sehr praxisorientiert und erprobt sind, sondern den Nutzer*innen z.B. Übersetzungen der Empfehlungen in über 40 Sprachen und einzelne Implementierungen als Open Source zur Verfügung stehen.

Reviewer 3: ```suggestion
Das Konsortium besteht aus einem fixen Team mit mehr als 60 Personen. Die vier Gastgeberorganisationen sind das 'MIT Computer Science and Artificial Intelligence Laboratory' (MIT/CSAIL) in den USA, das 'European Research Consortium for Informatics and Mathematics' (ERCIM) in Frankreich, die 'KEIO University' in Japan und die 'Beihang University' in China. Dazu kommen über 400 Mitgliederorganisationen aus mehr als 30 Ländern. Ergänzt wird die Arbeit dieser beiden Gruppen durch Beiträge aus der interessierten Web-Community, die an den Entwicklungs- und Diskussionsprozessen maßgeblich beteiligt ist. Ihre Mitarbeit ist ein Garant dafür, dass die Empfehlungen nicht nur sehr praxisorientiert und erprobt sind, sondern den Nutzer*innen z.B. Übersetzungen der Empfehlungen in über 40 Sprachen und einzelne Implementierungen als Open Source zur Verfügung stehen.
```
=> Nur fehlende einfache Anführungszeichen ergänzt.

b) <element attribut="wert" attribut2="wert"/>
```

## XML-Syntax

Jede XML-Annotation bedient sich dieser einfachen Bausteine. In der konkreten Anwendung sind jedoch besondere Regeln einzuhalten. U.a. syntaktische Vorschriften.

### Wohlgeformtheit

Die wichtigste Syntaxregel ist die Einhaltung der **Wohlgeformtheit** oder **'well-formedness'**. Sie bedeutet, dass **jedes geöffnete Element auch wieder geschlossen** werden muss, wie die hier angegebenen Elemente a und b. **Andernfalls** kann es sich nur um ein **leeres Element** handeln, wie das angeführte c-Element: 

```xml
<a>Inhalt</a> Inhalt <b>Inhalt</b> Inhalt ... <c/>
```

Sollten Elemente **verschachtelt** sein, so müssen sie in der **richtigen Reihenfolge geöffnet und geschlossen** werden. Dies gilt auch für den Fall, dass sich mehrere Elemente überlappen: 

_Falsch:_

```xml
<a>Inhalt <b>Inhalt</a> Inhalt</b>
```

_Richtig:_

```xml
<a>Inhalt <b>Inhalt</b></a> <b>Inhalt</b>
```

<SideNote type="note" title="Merke">
Die Regel der 'well-formedness' ist zwingend, d.h. **alle XML-Dokumente müssen wohlgeformt sein!**

- Geöffnete Elemente müssen wieder geschlossen werden oder leere Elemente sein. 

> - Bei verschachtelten bzw. sich überlappenden Elementen ist die richtige Reihenfolge beim Öffnen und Schließen von Elementen zu berücksichtigen.

Reviewer 3: ```suggestion
- Bei verschachtelten bzw. sich überlappenden Elementen ist die richtige Reihenfolge beim Öffnen und Schließen zu berücksichtigen.
```

</SideNote>

### Validität

Darüber hinaus kann ein XML-Dokument auch valide oder gültig sein. Was bedeutet das? 

Wie schon erwähnt, kann im Grunde jede*r mit XML seine eigene Markup-Sprache für bestimmte Zwecke definieren. Gleichzeitig ist es mit Blick auf kollaborative Arbeitsweisen und Open Science unerlässlich, die erzeugten Dokumente weitergeben und mehrfach nachnutzen zu können. **Wer also eine Sprache "erfindet", sollte nicht nur das Vokabular, sprich die Elemente, Attribute und Werte festlegen, sondern auch eine "Grammatik" – Schema genannt – schreiben**, in der definiert ist, was die Elemente bedeuten, in welcher Beziehung sie zueinander stehen, wo und wie sie verwendet werden dürfen. 

Obwohl diese Regel optional ist, empfiehlt es sich, sie einzuhalten – aus mehreren Gründen: 

- Ein Schema ist zum einen nützlich, damit man selbst überprüfen kann, ob man mehrere Dokumente desselben Typs einheitlich annotiert hat. 
- Zum anderen dokumentiert es die vorhandene Struktur und macht sie für Dritte nachvollziehbar, die u.U. gleichartige Dokumente auf dieselbe Weise annotieren möchten, oder Daten von anderen Wissenschaftler*innen nachnutzen und weiterverarbeiten wollen.

> Es gibt mehrere Möglichkeiten, ein Schema zu schreiben. Am Beginn der Entwicklung stand **['Document Type Definition'](https://www.w3schools.com/xml/xml_dtd.asp) (DTD)**. DTD wurden vom XML-Vorgängerstandard [Standard Generalized Markup Language (SGML)](https://www.w3.org/History/19921103-hypertext/hypertext/WWW/MarkUp/SGML.html) übernommen, ist allerdings in einer Sprache geschrieben, die selbst kein XML ist. Dies bedeutet einen Zusatzaufwand, da man diese Sprache erst erlernen muss. 

Reviewer 3: ```suggestion
Es gibt mehrere Möglichkeiten, ein Schema zu schreiben. Am Beginn der Entwicklung stand **['Document Type Definition'  (DTD)](https://www.w3schools.com/xml/xml_dtd.asp)**. DTD wurden vom XML-Vorgängerstandard **['Standard Generalized Markup Language' (SGML)](https://www.w3.org/History/19921103-hypertext/hypertext/WWW/MarkUp/SGML.html)** übernommen, ist allerdings in einer Sprache geschrieben, die selbst kein XML ist. Dies bedeutet einen Zusatzaufwand, da man diese Sprache erst erlernen muss. 
```

> Dieser Mangel konnte mit **[XML Schema Definition (XSD)](https://www.w3.org/TR/xmlschema11-1/)** im Jahr 2001 behoben werden – das ist ebenfalls eine W3C-Empfehlung und mittlerweile sehr weit verbreitet. 

Reviewer 3: ```suggestion
Dieser Mangel konnte mit **['XML Schema Definition' (XSD)](https://www.w3.org/TR/xmlschema11-1/)** im Jahr 2001 behoben werden – das ist ebenfalls eine W3C-Empfehlung und mittlerweile sehr weit verbreitet. 
```

> Daneben gibt es noch weitere Spezifikationen um XML-Strukturen zu definieren wie z.B. **[Regular Description for XML New Generation (Relax NG)](https://relaxng.org/)** und andere mehr. Da die zuletzt genannten in XML geschrieben sind, sind auch sie praktischerweise für Menschen und Maschinen lesbar.

Reviewer 3: ```suggestion
Daneben gibt es noch weitere Spezifikationen um XML-Strukturen zu definieren wie z.B. **['Regular Description for XML New Generation' (Relax NG)](https://relaxng.org/)** und andere mehr. Da die zuletzt genannten in XML geschrieben sind, sind auch sie praktischerweise für Menschen und Maschinen lesbar.
```

Neben den Beziehungen regelt ein Schema auch, **wie oft ein Element oder Attributwert vorkommen darf**. 

- Eindeutige ID-Nummern dürfen sinnvollerweise nur einmal vergeben werden. 
- Ebenso sollte in einem Geburtsregister jede Person nur ein Geburtsdatum haben.

Die Struktur eines Buches könnte in XML etwa wie folgt abgebildet werden: 

![XML Hierarchie eines Buches](images/xml-hierarchie_neu.png "XML-Struktur eines Buches. Bild: Ulrike Czeitschner, ACDH-CH")

Innerhalb des Wurzelelements befindet sich ein div-Element ("div" steht für 'division', d.h. Textabschnitt) mit dem Attribut `@type=book` für das Buch, das die Kindelemente Vorwort und mehrere Kapitel enthält, vielleicht gefolgt von einem Nachwort, eventuell einem Register und einem Impressum. Diese Kindelemente verhalten sich zueinander alle wie Geschwister und sind selbst Eltern der Titel-, Paragrafen- und Verszeilen-Elemente, die ebenfalls Geschwister sind und selbstverständlich auch wieder Kinder haben können. Die so strukturierten Inhalte spiegeln den Aufbau des Buches wider.

## Von XML zu TEI

> Das TEI-Dokumentenformat, herausgegeben von der 1987 gegründeten **[Text Encoding Initiative](https://tei-c.org/)**, basiert auf XML und ist mittlerweile der de-facto-Standard für die Kodierung von Texten und sprachlichen Informationen. Die derzeit empfohlene Version P5 wurde 2007 veröffentlicht und seitdem regelmäßig verbessert und aktualisiert. Das Regelwerk enthält mehr als 500 Elemente und ebenso zahlreiche Attribute, die in sachbezogenen [Modulen](https://tei-c.org/release/doc/tei-p5-doc/en/html/index.html) organisiert sind. Sie berücksichtigen Lyrik und Dramentexte ebenso wie Handschriften, kritische Apparate oder Wörterbücher. Darüber hinaus bietet die TEI Komponenten sowohl für die Schema-Generierung als auch -Anpassung an und stellt integrierte Stylesheets zur Verfügung.

Reviewer 3: ```suggestion
Das TEI-Dokumentenformat, herausgegeben von der 1987 gegründeten **['Text Encoding Initiative'](https://tei-c.org/)**, basiert auf XML und ist mittlerweile der de-facto-Standard für die Kodierung von Texten und sprachlichen Informationen. Die derzeit empfohlene Version P5 wurde 2007 veröffentlicht und seitdem regelmäßig verbessert und aktualisiert. Das Regelwerk enthält mehr als 500 Elemente und ebenso zahlreiche Attribute, die in sachbezogenen [Modulen](https://tei-c.org/release/doc/tei-p5-doc/en/html/index.html) organisiert sind. Sie berücksichtigen Lyrik und Dramentexte ebenso wie Handschriften, kritische Apparate oder Wörterbücher. Darüber hinaus bietet die TEI Komponenten sowohl für die Schema-Generierung als auch -Anpassung an und stellt integrierte Stylesheets zur Verfügung.
```
=> Nur fehlende einfache Anführungszeichen ergänzt.

> Der Begriff "Markup" stammt ursprünglich aus der Druckersprache und bedeutet "Textauszeichung". Ein Beispiel dafür wären Fahnenkorrekturen. Sie enthalten z.B. Fehlerberichtigungen und Anweisungen, was mit den angestrichenen Stellen zu tun ist – Buchstabenverdreher umstellen, Doppelungen löschen, Absätze aufteilen usw. Dabei werden Prozeduren vorgegeben, die früher der Setzer, heute der Grafiker mit dem Text zu vollziehen hat. Das könnte man als **"Prozedurales Markup"** bezeichnen.

Reviewer 3: ```suggestion
Der Begriff "Markup" stammt ursprünglich aus der Druckersprache und bedeutet "Textauszeichung". Ein Beispiel dafür wären Fahnenkorrekturen. Sie enthalten z.B. Fehlerberichtigungen und Anweisungen, was mit den angestrichenen Stellen zu tun ist – Buchstabenverdreher umstellen, Doppelungen löschen, Absätze aufteilen usw. Dabei werden Prozeduren vorgegeben, die früher der Setzer, heute der Grafiker mit dem Text zu vollziehen hat. Das könnte man als **Prozedurales Markup** bezeichnen.
```
=> Nur Anführungszeichen entfernt.

> In vielen Fällen ist das Ziel von solchen "Prozeduren" die Darstellung eines Dokuments in einer digitalen Druckvorlage. Wenn die Auszeichnung darauf abzielt, solche Darstellungsregeln zu definieren, spricht man vom **"Präsentationalen Markup"**.

Reviewer 3: ```suggestion
In vielen Fällen ist das Ziel von solchen "Prozeduren" die Darstellung eines Dokuments in einer digitalen Druckvorlage. Wenn die Auszeichnung darauf abzielt, solche Darstellungsregeln zu definieren, spricht man vom **Präsentationalen Markup**.
```
=> Nur Anführungszeichen entfernt.
